### PR TITLE
feat: introduce cli commands for the upgrade module

### PR DESCRIPTION
**Pull request by cmwaters**
Addresses: https://github.com/celestiaorg/celestia-app/issues/2875

This also adds an integration test to the upgrade module

### CODE REVIEW

File: x/upgrade/cli/tx.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func CmdTryUpgrade() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "try-upgrade",
+		Short: "Try to perform a software upgrade",
+		Long: `This command will submit a TryUpgrade message to tally all
+the signal votes. If a quorum has been reached, the network will upgrade
```
Comment (staheri14) [inline x/upgrade/cli/tx.go:68]: [Question] What indications should a user look for after executing this command to confirm that a network upgrade is in progress/going to happen? I am referring to this part of the command description "If a quorum has been reached, the network will upgrade to the signalled version at the following height."

**Review by evan-forbes — APPROVED** [review]
dope!

do you think its worth it to run the msgs through the anti handlers in some way? perhaps we could just add them to https://github.com/celestiaorg/celestia-app/blob/c5e178642f9854d774ea2cdd112f85d495f20032/app/test/std_sdk_test.go#L74 (and maybe rename that test)

it might be overly paranoid, but the reason being that we could remove the output to something like the getsigners method (or other critical piece) atm and all tests would pass.

File: x/upgrade/cli/query.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	"github.com/celestiaorg/celestia-app/x/upgrade/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/spf13/cobra"
+)
+
+// GetQueryCmd returns the CLI query commands for this module
+func GetQueryCmd() *cobra.Command {
+	// Group blob queries under a subcommand
```
[the remark this answers is not in the document]
Comment (rootulp) [reply x/upgrade/cli/query.go:15]: Nope. 
```suggestion
```

File: x/upgrade/cli/query.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	}
+
+	cmd.AddCommand(CmdQueryTally())
+
+	return cmd
+}
+
+func CmdQueryTally() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tally [version]",
```
Comment (rootulp) [inline x/upgrade/cli/query.go:31]: [nit] version is a required argument but the surrounding `[]` make it look like an optional argument.

```suggestion
		Use:   "tally version",
```

IDK if there's an authoritative source for command line argument help syntax. I think `tally <version>` or `tally version` are preferable over `tally [version]`

Ref: 
- https://developers.google.com/style/code-syntax
- https://stackoverflow.com/a/47026846/2813210

File: x/upgrade/cli/tx.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+	cmd.AddCommand(CmdSignalVersion())
+	cmd.AddCommand(CmdTryUpgrade())
+
+	return cmd
+}
+
+func CmdSignalVersion() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "signal [version]",
```
Comment (rootulp) [inline x/upgrade/cli/tx.go:33]:
```suggestion
		Use:   "signal version",
```

File: x/upgrade/integration_test.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+		Version:          2,
+	})
+	require.NoError(t, err)
+
+	res, err = app.UpgradeKeeper.VersionTally(goCtx, &types.QueryVersionTallyRequest{
+		Version: 2,
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, 1, res.VotingPower)
+	require.EqualValues(t, 0, res.ThresholdPower)
```
Comment (rootulp) [inline x/upgrade/integration_test.go:48]: [optional] this threshold power seems weird b/c it makes me think that no votes are needed to trigger an upgrade.
 Is it possible for this test case to have a non-zero threshold power potentially by adding another validator with some voting power?

File: x/upgrade/tally_test.go [hunk 6]
```diff
@@ -167,6 +167,18 @@ func TestTallyingLogic(t *testing.T) {
...
+func TestEmptyStore(t *testing.T) {
+	upgradeKeeper, ctx, _ := setup(t)
+	goCtx := sdk.WrapSDKContext(ctx)
+
+	res, err := upgradeKeeper.VersionTally(goCtx, &types.QueryVersionTallyRequest{
+		Version: 2,
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, 0, res.VotingPower)
+	require.EqualValues(t, 120, res.TotalVotingPower)
```
[the remark this answers is not in the document]
Comment (rootulp) [reply x/upgrade/tally_test.go:179]: `120` is the sum of voting powers in:

```
		validators: map[string]int64{
			testutil.ValAddrs[0].String(): 40,
			testutil.ValAddrs[1].String(): 1,
			testutil.ValAddrs[2].String(): 60,
			testutil.ValAddrs[3].String(): 19,
		},
```

from `setup` below

**Review by rootulp — APPROVED** [review]
LGTM

File: x/upgrade/cli/query.go (hunk 3 above)
Comment (cmwaters) [reply x/upgrade/cli/query.go:31]: Yeah good call

File: x/upgrade/integration_test.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/celestiaorg/celestia-app/x/upgrade/types"
+	"github.com/stretchr/testify/require"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tmlog "github.com/tendermint/tendermint/libs/log"
+	tmtypes "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmversion "github.com/tendermint/tendermint/proto/tendermint/version"
+)
+
+func TestUpgradeIntegration(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (cmwaters) [reply x/upgrade/integration_test.go:17]: Sure thing

**Review by evan-forbes — DISMISSED** [review]

**Review by rootulp — APPROVED** [review]
🚢 great work!

**Review by evan-forbes — APPROVED** [review]

**Review by rootulp — APPROVED** [review]

**Pull request merged**